keyboards/strech/soulstone/keymaps/default/keymap.c

```diff
@@ -0,0 +1,54 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,             KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_DEL,   KC_BSPC,
+        KC_TAB,   KC_A,     KC_S,     KC_D,     KC_F,     KC_G,             KC_H,     KC_J,     KC_K,     KC_L,     KC_QUOT,  KC_ENT,
+        KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,             MO(3),    KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,
+        KC_LCTL,  KC_LALT,                      MO(2),    MO(1),            KC_SPC,   KC_RALT,                                KC_RGUI,  KC_RCTL
+    ),
+    [1] = LAYOUT(
+        KC_GRV,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_UP,    KC_TRNS,  KC_MUTE,  KC_VOLD,  KC_VOLU,
+        KC_CAPS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_LEFT,  KC_DOWN,  KC_RGHT,  KC_SCLN,  KC_BSLS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    ),
+    [2] = LAYOUT(
+        KC_TRNS,  KC_1,     KC_2,     KC_3,     KC_4,     KC_5,             KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P4,    KC_P5,    KC_P6,    KC_LBRC,  KC_RBRC,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P1,    KC_P2,    KC_P3,    KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_PENT,  KC_PDOT,                                KC_TRNS,  KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,            KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    )
+};
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    if (get_highest_layer(state) == 1) {
+        writePinHigh(B3);
+    } else {
+        writePinLow(B3);
+    }
```

Comment:
This might be better, as it will enable the LED if layer 1 is enabled, no matter if it's the highest layer or not.  Which sounds like it may be closer to what you want? 
```suggestion
    writePin(B3, layer_state_cmp(state, 1));
```

Reply:
I've replied here https://github.com/qmk/qmk_firmware/pull/22093#discussion_r1334962631